app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -158,6 +164,20 @@ class BrowserWebViewClient @Inject constructor(
         try {
             Timber.v("shouldOverride webViewUrl: ${webView.url} URL: $url")
             webViewClientListener?.onShouldOverride()
+
+            if (runBlocking {
```

Comment:
proposal: remove the `suspend` for shouldOverrideUrlLoading, and move the runBlocking inside the implementation of that method. That way we can control where exactly we need it and it's an implementation detail of that method.